src/test/java/com/epam/tc/hw/mobile/scenarios/NativeMobileTests.java

```diff
@@ -6,24 +6,37 @@
 import org.testng.annotations.Test;
 import com.epam.tc.hw.mobile.setup.BaseTest;
 
+import java.util.Objects;
+
 public class NativeMobileTests extends BaseTest {
 
     @DataProvider(name = "nativeEpamTest")
     public Object[][] nativeEpamTest() {
-        return new Object[][] {
+        return new Object[][]{
                 {"emailName@epam.com", "EPAM", "asnjalf195", "BudgetActivity"}};
     }
 
     @Test(groups = {"native"}, dataProvider = "nativeEpamTest",
             description = "This simple test register in EpamTestApp and sign in as registered user")
     public void registerNativeTest(String email, String name, String password, String endPoint) {
+        String platform = getDriver().getPlatformName();
         NativePageObject nativePageObject = new NativePageObject(getDriver());
-        nativePageObject.registerUser(email, name, password);
-        nativePageObject.loginUser(email, name);
 
-        Assert.assertEquals(endPoint, nativePageObject.getPageName(),
-                "failure - Failed to login in app");
+        if (Objects.equals(platform, "Android")) {
```

Comment:
Please, move all the logic that is not connected with business logic = test logic outside of the test. What platform should be used must be defined on another level, for ex. BaseTest or some service that can be used by BaseTest.